FF103 updates for scroll-snap-stop

dipikabh opened the pull request:
#### Motivation
<!-- ❓ Why are you making this change? Help us understand how your changes help readers. -->
`scroll-snap-stop` is now available in Firefox. The CSS property [scroll-snap-stop](https://developer.mozilla.org/en-US/docs/Web/CSS/scroll-snap-stop) is already documented.

<!-- 👀 Thanks for opening a PR! Read comments like this one to get your PR merged faster. -->
#### Summary
<!-- ✍️ In a sentence or two, describe your changes -->
- In the [Basic concepts page](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Scroll_Snap/Basic_concepts), removed the sentence "This property has fewer browser implementations than the rest of the specification." in the "scroll-snap-stop property" section. Other language changes done on the page include:
  - Made the headings parallel in language and information
  - Edits for commas
  - Removed instances of "I"
- In the [scroll-snap-stop](https://developer.mozilla.org/en-US/docs/Web/CSS/scroll-snap-stop) property page:
  - Updated position of "Section" heading
  - In the "Examples" section:
    - Changed the order to HTML and then CSS for consistency with other CSS property pages
    - Reduced the HTML bit of the example to retain only the relevant portions that demonstrate `scroll-snap-stop`
  - Added more links to "See also"
- PR for BCD update: https://github.com/mdn/browser-compat-data/pull/17122

#### Related issues
<!-- 🔨 If applicable, use "Fixes #XYZ" -->
Doc issue tracker: https://github.com/mdn/content/issues/17469
https://bugzilla.mozilla.org/show_bug.cgi?id=1312165

#### Supporting details
<!-- 🔗 Link to supporting information, such as bug trackers, source control, release notes, and vendor docs. -->
https://drafts.csswg.org/css-scroll-snap/#scroll-snap-stop

#### Metadata
<!-- ✅ Check a box if applicable, like this: [x]

This PR…
-->
- [ ] Adds a new document
- [X] Rewrites (or significantly expands) a document
- [ ] Fixes a typo, bug, or other error

<!-- 👷‍♀️ After submitting, review the results of the "Checks" tab! -->

dipikabh committed:
FF103 updates for scroll-snap-stop

estelle commented on files/en-us/web/css/css_scroll_snap/basic_concepts/index.md:
```diff
@@ -11,55 +11,60 @@ tags:
 {{EmbedGHLiveSample("css-examples/scroll-snap/scroll-margin.html", '100%', 700)}}
 
-## The scroll-snap-stop property
+## Using scroll-snap-stop
```
```suggestion
## Using `scroll-snap-stop`
```

teoli2003 replied:
> 
The style guide explicitly forbids the use of backticks on headings. (I'm not really happy about it). But we are working into removing the few remaining.

Let's not add new ones.

estelle replied:
Ah, thanks, i thought it was the opposite. D'oh.

estelle commented on files/en-us/web/css/scroll-snap-stop/index.md:
```diff
@@ -105,12 +136,21 @@ This example is duplicated from {{cssxref("scroll-snap-type")}} with minor varia
   flex: none;
 }
 
+/* defining scroll-snap stops */
```
If we remove these, the example still works because of the mandatory and proximity I think, so we should likely edit those so that this section has an impact.

estelle replied:
i don't know anything about this property, so researching.....

estelle replied:
I created a codepen https://codepen.io/estelle/pen/RwMZBRQ that we may want to use (or use half of). I changed the width and heights to 90% so it's more apparent that they need to scroll, and added the value of the property in perm text above each example. I am able to scroll over the middle divs even with the "always" version, so don't fully get it, but I think using all 8, or the top 4 or bottom 4 is better because mandatory is not always with always and proximity isn't always matched with normal.

estelle commented on files/en-us/web/css/scroll-snap-stop/index.md:
```diff
@@ -121,95 +161,28 @@ This example is duplicated from {{cssxref("scroll-snap-type")}} with minor varia
 
 .container > div:nth-child(odd) {
   background-color: #87CCEA;
-}
```
the closing } needs to be put back in.

teoli2003 commented on files/en-us/web/css/css_scroll_snap/basic_concepts/index.md:
```diff
@@ -11,55 +11,60 @@ tags:
 {{EmbedGHLiveSample("css-examples/scroll-snap/scroll-padding-sticky.html", '100%', 700)}}
 
-## Margins on scroll children
+## Using scroll-margin
```
See above.

teoli2003 commented on files/en-us/web/css/css_scroll_snap/basic_concepts/index.md:
```diff
@@ -11,55 +11,60 @@ tags:
 {{EmbedGHLiveSample("css-examples/scroll-snap/align.html", '100%', 700)}}
 
-## Padding the scroll container
+## Using scroll-padding
```
See above.

dipikabh committed:
Fix review comments

dipikabh commented:
Thanks a lot for reviewing this, Estelle. Much appreciate you rewriting the code example. 🙏 

> I created a codepen https://codepen.io/estelle/pen/RwMZBRQ that we may want to use (or use half of). I changed the width and heights to 90% so it's more apparent that they need to scroll, and added the value of the property in perm text above each example.

I've used a lot of your code from codepen. I've also hidden the CSS for appearance of the examples as you suggested. I like the idea of text above the examples.

> I was able to scroll over the middle divs even with the "always" version, so changed it to always/normal on even divs only, and the effect is now noticeable ONLY on mandatory.

To simplify this, I've retained only the `mandatory` examples since those are the ones that are more noticeable than the `proximity` ones. I've reworded the text under the headings "Setting different snap stops" and "Result" to explain this.

estelle commented on files/en-us/web/css/scroll-snap-stop/index.md:
```diff
@@ -47,169 +47,139 @@ scroll-snap-stop: unset;
 #### Result
 
-{{EmbedLiveSample("Snapping_in_different_axes", "100%", "1630")}}
+Scroll from left to right and from top to bottom in the X and Y boxes below, respectively. In the X and Y boxes where the `scroll-snap-stop` property is set to `always`, the scrolling is forced to stop at the snap point even when you scroll fast. However, with the `scroll-snap-stop` property set to `normal`, the snap points are skipped when you scroll fast.
```
```suggestion
Scroll from left to right and from top to bottom in the X and Y boxes below, respectively. In the X and Y boxes where the `scroll-snap-stop` property is set to `always`, the scrolling is forced to stop at the snap point even when you scroll fast. However, with the `scroll-snap-stop` property set to `normal`, the snap points are skipped when you scroll fast.

Be selective as to which items gets `always` as setting them all to `always` means none are more important than others. In our example we set every other child; in this case, the even numbered items.
```

estelle reviewed (changes requested):
while it isn't in the specification, we should include the recommendation to only set a few items to `always`. If they're all set to `always`, there is no "more important to stop on this item than others" and the effect of always is nullified. I suggested a way of saying it. Your suggestion will likely be better than mine ;)

dipikabh committed:
Fix review comments part 2

dipikabh commented:
`scroll-snap-stop: normal` is like the default behavior when no `scroll-snap-stop` is set up, so alternating this value for the child elements would not impact difference in scroll behavior across these elements.

I am not quite sure about making that recommendation. For one, the kind of scroll examples I have seen are horizontal image carousels or vertical section scrolls on home page. In both the situations, one would most likely go for `scroll-snap-stop: always` on all child elements, if at all setting that property, to give a uniform scroll experience. I am not aware of situations where one would need to consider the importance of an item in a scroll container to decide on `scroll-snap-stop`.

But I think we should still demo how to selectively enable `always` using your example. I've added two separate instances to show`scroll-snap-stop: always` on odd elements only and on even elements only. This will hopefully bring out the contrast among the three scenarios (always, always odd, always even).

Please let me know what you think.

estelle reviewed (approved):
w00t!
